simulation/traffic_simulator/src/entity/entity_manager.cpp

```diff
@@ -254,13 +254,17 @@ auto EntityManager::getLongitudinalDistance(
 {
   if (!include_adjacent_lanelet) {
     auto forward_distance = hdmap_utils_ptr_->getLongitudinalDistance(
-      static_cast<LaneletPose>(from), static_cast<LaneletPose>(to));
+      from.getNonCanonicalizedLaneletPose(), to.getNonCanonicalizedLaneletPose());
 
     auto backward_distance = hdmap_utils_ptr_->getLongitudinalDistance(
-      static_cast<LaneletPose>(to), static_cast<LaneletPose>(from));
+      to.getNonCanonicalizedLaneletPose(), from.getNonCanonicalizedLaneletPose());
```

Comment:
We believe that after the LaneletPose is Canonicalized, the Lanelet Pose before Canonicalization should no longer be needed.
The fix included in the /fix/longitudinal_distance branch fixes a bug that was caused by the Lanelet Pose before it was Canonicalized and allows the NonCanonicalizedLaneletPose to be used with the If you use NonCanonicalizedLaneletPose to calculate longitudinal distance, the bug will not be corrected.